src/lsst/cmservice/web_app/app.py

```diff
@@ -96,16 +95,15 @@ async def get_campaigns(
 ) -> HTMLResponse:
     try:
         async with session.begin():
-            production_list = {}
-            productions = await db.Production.get_rows(session)
-            for p in productions:
-                children = await p.children(session)
-                production_campaigns = []
-                for c in children:
-                    campaign_details = await get_campaign_details(session, c)
-                    production_campaigns.append(campaign_details)
-                production_list[p.name] = production_campaigns
-
+            production_list: dict[str, list] = {}
+            campaigns = await get_all_campaigns(session)
+            for campaign in campaigns:
+                production_name = campaign.fullname.split("/")[0]
```

Comment:
I think `production_name` will eventually become part of the campaign object, so moving this string-parsing into `get_campaign_details` would make it easier to adjust in the future.